src/models/contact.js

```diff
@@ -229,11 +230,13 @@ export default class Contact {
 	 * Return the photo usable url
 	 * We cannot fetch external url because of csp policies
 	 *
-	 * @readonly
 	 * @memberof Contact
 	 */
-	get photoUrl() {
+	async getPhotoUrl() {
```

Comment:
There are ways to have an async getter :)

Reply:
Yes there are but it feels like that's not what getters are supposed to do. I would rather have this operation in a traditional method

Reply:
Hum, that's also a fair point.
That would solve the contact change and avoid weird watchers for the `ContactDetailsAvatar.vue`

Reply:
The watcher is weird indeed. But even if I have an async getter then I can't use a computed props because Vue computed props are not async. One needs a custom plugin for that and I'm not sure if that is not creating more problems. After all template rendering is done sync. So there also needs to be some kind of default value for an async prop before it got evaluated. It's … weird.

Reply:
What about something like that?

```js
  get photoUrl() {
    return (async () => {
      try {
        ...
        const cleanSvg = await sanitizeSVG(imageSvg)
        ...
      } catch(e) {
        return 0;
      }
    })();
```

Reply:
:ping_pong: it still doesn't make sense to use async code inside sync code. you won't be able to evaluate the result of the promise

Reply:
Talked with @st3iny about this. The code above won't work. We need the async method, an initial value and a watcher for changes.